server/src/core/server/services/users/emailPremodFilter.ts

```diff
@@ -121,26 +95,43 @@ export const shouldPremodDueToLikelySpamEmail = async (
     return false;
   }
 
+  if (!user.email) {
```

Comment:
fyi, consolidated some checks for user email, email split length, etc., into here. that way each premod filter check can focus on its more specific check.